lib/main_common.pm

```diff
@@ -1600,44 +1601,86 @@ sub load_filesystem_tests {
+    }
+}
+
 sub load_wicked_tests {
     loadtest 'wicked/before_test';
+    my $folder;
+    if (get_var('IS_WICKED_REF')) {
+        $folder = 'ref';
+        load_wicked_barrier();
+    } else {
+        $folder = 'sut';
+    }
     if (check_var('WICKED', 'basic')) {
-        loadtest 'wicked/basic/t01_basic';
-        loadtest 'wicked/basic/t02_static_addresses_legacy';
-        loadtest 'wicked/basic/t03_static_addresses_xml';
-        loadtest 'wicked/basic/t04_dynamic_addresses_legacy';
-        loadtest 'wicked/basic/t05_dynamic_addresses_xml';
-        loadtest 'wicked/basic/t06_static_routes_legacy';
-        loadtest 'wicked/basic/t07_static_routes_xml';
+        for my $test (get_wicked_tests()) {
+            loadtest 'wicked/basic/' . $test;
+        }
     }
     elsif (check_var('WICKED', 'advanced')) {
-        my $f = get_var('IS_WICKED_REF') ? 'ref' : 'sut';
-        loadtest 'wicked/advanced/' . $f . '/t01_gre_tunnel_legacy';
-        loadtest 'wicked/advanced/' . $f . '/t02_gre_tunnel_xml';
-        loadtest 'wicked/advanced/' . $f . '/t03_sit_tunnel_legacy';
-        loadtest 'wicked/advanced/' . $f . '/t04_sit_tunnel_xml';
-        loadtest 'wicked/advanced/' . $f . '/t05_ipip_tunnel_legacy';
-        loadtest 'wicked/advanced/' . $f . '/t06_ipip_tunnel_xml';
-        loadtest 'wicked/advanced/' . $f . '/t07_tun_interface_legacy';
-        loadtest 'wicked/advanced/' . $f . '/t08_tun_interface_xml';
-        loadtest 'wicked/advanced/' . $f . '/t09_tap_interface_legacy';
-        loadtest 'wicked/advanced/' . $f . '/t10_tap_interface_xml';
-        loadtest 'wicked/advanced/' . $f . '/t11_bridge_interface_legacy';
-        loadtest 'wicked/advanced/' . $f . '/t12_bridge_interface_xml';
+        for my $test (get_wicked_tests()) {
+            loadtest 'wicked/advanced/' . $folder . '/' . $test;
+        }
     }
     elsif (check_var('WICKED', 'startandstop')) {
-        my $f = get_var('IS_WICKED_REF') ? 'ref' : 'sut';
-        loadtest 'wicked/startandstop/' . $f . '/t01_standalone_card_ifdown_ifreload';
-        loadtest 'wicked/startandstop/' . $f . '/t02_bridge_ifreload';
-        loadtest 'wicked/startandstop/' . $f . '/t03_bridge_ifup_ifreload';
-        loadtest 'wicked/startandstop/' . $f . '/t04_bridge_ifup_remove_all_config_ifreload';
-        loadtest 'wicked/startandstop/' . $f . '/t05_bridge_ifup_remove_one_config_ifreload';
-        loadtest 'wicked/startandstop/' . $f . '/t06_bridge_ifdown_create_new_config_ifreload_ifdown_ifup';
-        loadtest 'wicked/startandstop/' . $f . '/t07_bridge_ifdown_remove_one_config_ifreload_ifdown_ifup';
-        loadtest 'wicked/startandstop/' . $f . '/t08_sit_tunnel_ifdown';
-        loadtest 'wicked/startandstop/' . $f . '/t09_openvpn_tunnel_ifdown';
-        loadtest 'wicked/startandstop/' . $f . '/t10_vlan_ifup_all_ifdown_one_card';
+        for my $test (get_wicked_tests()) {
+            loadtest 'wicked/startandstop/' . $folder . '/' . $test;
+        }
```

Comment:
could we do something like

    for my $test (get_wicked_tests()) {
        loadtest 'wicked/' .  get('WICKED') . '/' . $folder . '/' . $test;
    }

and if we want to keep the check for "unhandled WICKED test selection" we can still do it before the for-loop.

Reply:
yes we can do it, when basic is changed.